Sources/LanguageServerProtocol/SupportTypes/RegistrationOptions.swift

```diff
@@ -156,6 +156,30 @@ public struct InlayHintRegistrationOptions: RegistrationOptions, TextDocumentReg
   }
 }
 
+/// Describe options to be used when registering for pull diagnostics. Since LSP 3.17.0
+public struct DiagnosticRegistrationOptions: RegistrationOptions, TextDocumentRegistrationOptionsProtocol {
+  public var textDocumentRegistrationOptions: TextDocumentRegistrationOptions
+  public var diagnosticOptions: DiagnosticOptions
+  
+  public init(
+    documentSelector: DocumentSelector? = nil,
+    diagnosticOptions: DiagnosticOptions
+  ) {
+    textDocumentRegistrationOptions = TextDocumentRegistrationOptions(documentSelector: documentSelector)
+    self.diagnosticOptions = diagnosticOptions
+  }
+
+  public func encodeIntoLSPAny(dict: inout [String: LSPAny]) {
+    textDocumentRegistrationOptions.encodeIntoLSPAny(dict: &dict)
+
+    dict["interFileDependencies"] = .bool(diagnosticOptions.interFileDependencies)
+    dict["workspaceDiagnostics"] = .bool(diagnosticOptions.workspaceDiagnostics)
+    if let workDoneProgress = diagnosticOptions.workDoneProgress {
+      dict["workDoneProgress"] = .bool(workDoneProgress)
+    }
```

Comment:
I think you also need to encode `documentSelector` (from `TextDocumentRegistrationOptions` in the LSP spec) and `id` (from `StaticRegistrationOptions` in the LSP spec) here.

Reply:
`documentSelector` is already added to the dict by `textDocumentRegistrationOptions.encodeIntoLSPAny(dict: &dict)` at least.

Reply:
Whoops, and I apparently approved and merged the PR, forgotting that I still had a comment. If you could create a follow-up PR, that would be great.